File: #1143.longest-common-subsequence.cpp
class Solution {
public:
    int longestCommonSubsequence(string text1, string text2) {

        int row =text1.length();
        int col = text2.length();
        vector<vector<int>> v(row + 1,vector<int>(col+1));
        for (int i = 1; i <= row; i++) {
            for (int j = 1; j <= col; j++) {
                v[i][j] = text1[i-1] == text2[j-1] ?
                    v[i - 1][j-1] + 1
                    : std::max(v[i - 1][j], v[i][j - 1]);
            }
        }
        return v[row][col];
    }
};

File: #1143.longest-common-subsequence.js
/**
 * @param {string} text1
 * @param {string} text2
 * @return {number}
 */
var longestCommonSubsequence = function(text1, text2) {
    const length1 = text1.length;
    const length2 = text2.length;
    let dp = Array(length1 + 1)
        .fill(0).map(() => Array(length2 + 1)
        .fill(0));
    for (let i = length1 - 1; i >=0; i--) {
        for (let j = length2 - 1; j >= 0; j--) {
            text1[i] === text2[j] ?
                dp[i][j] = 1 + dp[i + 1][j + 1] :
                dp[i][j]= Math.max(dp[i][j + 1], dp[i + 1][j]);
        }
    }
    return dp[0][0];
};
